elasticdl/python/worker/worker.py

```diff
@@ -441,4 +482,32 @@ def run(self):
         elif self._job_type == JobType.EVALUATION_ONLY:
             self._evaluate_only()
         else:
-            self._train_and_evaluate()
+            if self.custom_training_loop:
```

Comment:
Why do we only execute allreduce if custom_training_loop is `true`?

Reply:
It is very difficult for users to define the training loop using PS in ElasticDL now.

Reply:
We can add more comments here.